### ISSUE TITLE
Badge Request: GitHub Package Registry - Version Badge 

### ISSUE THREAD

**Issue by calebcartwright** [badges/shields#4183]
:clipboard: **Description**
Download count badges for packages hosted on the GitHub Package Registry

- Which service is this badge for e.g: GitHub Package Registry 
- What sort of information should this badge show? Download counts, in whichever intervals the GitHub API makes available

![](https://img.shields.io/badge/downloads-10k-brightgreen)

:link: **Data**

- Is there a public API?  Yes
- Does the API requires an API key? Auth should be used like our other GH services, specifically leveraging our [BaseGraphQlService](https://github.com/badges/shields/blob/master/core/base-service/base-graphql.js). The [RegistryPackageStatistics](https://developer.github.com/v4/object/registrypackagestatistics/) will be of use
- Link to the API documentation. https://developer.github.com/v4/

:microphone: **Motivation**
Because people want them! 😄 The motivation/benefits for the version badges for the GH Package registry is the same as the other registries we support

Related: #3608 and #4071

**Comment by crimson-med** [comment]
Any update on this? I have recently been using Github Packages and would love to display version and download.

**Comment by calebcartwright** [comment]
Thanks for the bump @crimson-med! This, and other GitHub Package related badges are blocked on https://github.com/badges/shields/issues/4169

tl;dr - The Shields.io service leverages a pool of tokens that are used to retrieve data from the API for our GH badges, but none of those tokens have the requisite permissions needed to retrieve _Package_ data from the GH API. It's a non-trivial problem, but one I think we should focus some attention on (would encourage folks to track #4169 for more information)

**Comment by crimson-med** [comment]
@calebcartwright Thank you for the update I have subscribed to #4169 and hopefully a solution will come to light soon.

**Comment by xavisavvy** [comment]
Adding my comment here to watch this thread. This would indeed be a useful addition.

**Comment by Devwarlt** [comment]
I'm looking forward for this addition!

**Comment by malliapi** [comment]
bump !

**Comment by ngosang** [comment]
bump

**Comment by calebcartwright** [comment]
Folks, we post updates whenever they are available, so please refrain repeated "bump" comments which are not particularly helpful. You should instead use the reaction features to indicate your support for this enhancement.

If this is something you're particularly interested in, and would like to see it delivered more quickly than the maintainer team has bandwidth to do, then please consider submitting a PR

**Comment by chris48s** [comment]
While I agree with the general thrust of this (bump comments are unhelpful - we'll update if there's any progress), I think this needs clarifying. It is a bit misleading to say:

> If this is something you're particularly interested in, and would like to see it delivered more quickly than the maintainer team has bandwidth to do, then please consider submitting a PR

on this particular issue. There is already an open PR for this badge https://github.com/badges/shields/pull/4184 and the feature is blocked on https://github.com/badges/shields/issues/4169 which will require substantial input from the core team even if someone in the community **is** super-keen to help us develop a new OAuth/marketplace app to help with that. Obviously if someone is super-keen to help with that it would be a welcome contribution but it is a big piece of work and some design decisons are still ambiguous. This is not a typical "if you want the badge, submit a PR to add it - we've got a tutorial" situation.

**Comment by calebcartwright** [comment]
> It is a bit misleading to say
> There is already an open PR for this badge #4184 and the feature is blocked on #4169

I'm aware of the existing PRs and Issues (already shared within this thread) and stand by my previous comment and others on this thread, so we may have to agree to disagree on this one.

There's a body of work to be done, and while I it's absolutely true that there are a few decisions for the core team, I do not believe that 100% (or even a significant percentage) of the code needs to be typed by a maintainer nor do I believe the ability to _start_ making progress is contingent upon all of those decisions being finalized. 

I actually thinking that having someone (maintainer and/or community) start the work would help inform and accelerate those decisions and the ultimate resolution, even if some of that work only serves to prove a potential path infeasible. 

> This is not a typical "if you want the badge, submit a PR to add it - we've got a tutorial" situation.

I don't think anyone's said this, not even implicitly. I suppose I could've worded my comment differently, but the gist is that I think the community could help accelerate this.

**Comment by chris48s** [comment]
> we may have to agree to disagree on this one

We don't disagree, but IMO it needed clarifying what we would actually need help with to move this (and other things) forward.. Remember most of the people who have subscribed to this issue don't have all the context we do.

**Comment by CAMOBAP** [comment]
It will be really good to have support for this

**Comment by ronaldtse** [comment]
Agree!

**Comment by RosieWatson** [comment]
Are there anymore updates on this?

**Comment by calebcartwright** [comment]
> Are there anymore updates on this?

No, not really. Please see prior comment https://github.com/badges/shields/issues/4183#issuecomment-750918398

We have made some changes to the backend infrastructure (notably the mechanism we use to store the github tokens) which does open up some doors in design discussions.

However, there continues to be no substantive updates and the latest status on this issue is already reflected in the above comments.

We will happily share such updates here when they occur, but in the meantime I want to reiterate that the absence of explicit updates posted by the maintainers here means that there are no updates.

**Comment by ryansonshine** [comment]
> > Are there anymore updates on this?
> 
> No, not really. Please see prior comment [#4183 (comment)](https://github.com/badges/shields/issues/4183#issuecomment-750918398)
> 
> We have made some changes to the backend infrastructure (notably the mechanism we use to store the github tokens) which does open up some doors in design discussions.
> 
> However, there continues to be no substantive updates and the latest status on this issue is already reflected in the above comments.
> 
> We will happily share such updates here when they occur, but in the meantime I want to reiterate that the absence of explicit updates posted by the maintainers here means that there are no updates.

![](https://media1.tenor.com/m/VKU7xwwZJ_EAAAAC/dumb-and-dumber-lloyd.gif)
